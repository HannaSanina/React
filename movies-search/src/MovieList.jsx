import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import Image1 from './images/1.jpg'
import ErrorBoundary from './ErrorBoundary';
import './App.css';

const MoviePreview = ({ imageId }) => {
    return <span className="preview-container">
        <img className="movie" src={imageId} alt="Movie" />
        <div className="preview">Film title <button>Order</button></div>
    </span>
}

class MovieList extends React.Component {
    constructor(props) {
        super(props);
        this.state = { hasError: false };
    }

    render() {
        return <div>
            <div className="result-stats">
                <span> # Movies found</span>
                <span> Sort by</span>
            </div>  
            <ErrorBoundary>
                <div className="results-movie">
                    <MoviePreview imageId={Image1} />
                    <MoviePreview imageId={Image1} />
                    <MoviePreview imageId={Image1} />
                    <MoviePreview imageId={Image1} />
                    <MoviePreview imageId={Image1} />
                    <MoviePreview imageId={Image1} />
                </div>
            </ErrorBoundary>
        </div>;
    }
}
export default MovieList;