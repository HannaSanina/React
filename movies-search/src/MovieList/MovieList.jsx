import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import ErrorBoundary from './ErrorBoundary';
import MoviePreview from '../MoviePreview/MoviePreview';
import styles from './movieList.css'

export default class MovieList extends React.Component {
    render() {
        return <div>
            <ErrorBoundary>
                <div className={styles.resultsMovie} >
                    {this.props.movies.length ? this.props.movies.map(item => {
                        return <MoviePreview key={item.id} data={item} />
                    }) : ""}
                </div>
            </ErrorBoundary>
        </div>;
    }
}