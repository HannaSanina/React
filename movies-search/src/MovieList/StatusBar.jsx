import React from 'react';
import styles from './movieList.css';
import { Button } from 'reactstrap';

const filters = ['release_date', 'vote_count'];

export default class StatusBar extends React.Component {
    sortChange = this.sortChange.bind(this);

    state = { isToggleOn: true };

    sortChange() {
      this.setState(prevState => ({ isToggleOn: !prevState.isToggleOn }));
      this.props.onSortChange(this.state.isToggleOn ? filters[0] : filters[1]);
    }

    render() {
      return (
        <div className={styles.resultStats}>
          <span>
            {this.props.result}
            {' '}
Movies found
          </span>
          <span className={styles.sortOption}>
            {' '}
Sort by
            <Button color="primary" onClick={e => this.sortChange(e)}>
              <span className="buttonLabel">
                {' '}
                {this.state.isToggleOn ? 'Release date' : 'Rating'}
              </span>
            </Button>
          </span>
        </div>
      );
    }
}
