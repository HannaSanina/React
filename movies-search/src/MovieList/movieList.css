
.resultStats {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: 10px;
    background-color: lightblue;
}

.resultsMovie {
    display: flex;
    flex-flow: row wrap;
    justify-content: space-around;
}

.resultsMovie>span {
    width: 300px;
    padding-bottom: 20px;
    padding-top:10px;
}

.sortOption {
    display: flex;
    justify-content: space-between;
    width: 160px;
}