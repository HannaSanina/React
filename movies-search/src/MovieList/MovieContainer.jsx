import React from "react";
import MovieList from "./MovieList";
import StatusBar from "./StatusBar";
import PropTypes from 'prop-types';
import { fetchAllMovies, SORT_MOVIES, GET_ALL } from '../actions.jsx'
import { connect } from 'react-redux';
import styles from './movieList.css'

class MovieListContainer extends React.Component {
    state = { isToggleOn: true, sortOption: filters[0] };

    handleSortChange = this.handleSortChange.bind(this);

    componentDidMount() {
        const { store } = this.context;
        this.unsubscribe = store.subscribe(() => this.forceUpdate());
        this.props.fetchData()
    }

    componentWillUnmount() {
        this.unsubscribe();
    }

    handleSortChange(filterText) {
        const { store } = this.context;
        store.dispatch({ type: SORT_MOVIES, field: filterText });
    }

    render() {
        if (this.props.error) {
            return <p>Sorry! There was an error loading the items</p>;
        }

        const { store } = this.context;
        return <div>
            <StatusBar
                onSortChange={this.handleSortChange}
                result={store.getState().length} />
            <MovieList movies={store.getState()} />
        </div>;
    }
}

MovieListContainer.contextTypes = {
    store: PropTypes.object
}

const filters = ["release_date", "vote_count"];

const mapStateToProps = (state) => {
    return {
        data: state.data,
        error: state.error
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        fetchData: () => dispatch(fetchAllMovies())
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(MovieListContainer);