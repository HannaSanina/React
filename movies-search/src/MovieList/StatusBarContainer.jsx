import React from 'react';
import { connect } from 'react-redux';
import StatusBar from './StatusBar';
import { SORT_MOVIES } from '../actions';

class StatusBarContainer extends React.Component {
    handleSortChange = this.handleSortChange.bind(this);

    handleSortChange(filterText) {
      this.props.sortMovies(filterText);
    }

    render() {
      return (
        <div>
          <StatusBar
            onSortChange={this.handleSortChange}
            result={this.props.movies.length}
          />
        </div>
      );
    }
}

const mapStateToProps = state => ({ movies: state.movies });
const mapDispatchToProps = dispatch => ({
  sortMovies: filterText => dispatch({ type: SORT_MOVIES, field: filterText }),
});

export default connect(mapStateToProps, mapDispatchToProps)(StatusBarContainer);
