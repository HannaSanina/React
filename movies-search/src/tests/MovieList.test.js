import React from 'react';
import renderer from 'react-test-renderer';
import MovieList from '../MovieList/MovieList';

describe('<MovieList/>', () => {
    it('should be rendered correctly', () => {
        const component = renderer.create(
            <MovieList movies={movieDetails}/>
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
})

const movieDetails = [
    {
        "id": 0,
        "picture": "https://imgc.allpostersimages.com/img/posters/titanic_u-L-F4S6CQ0.jpg?src=gp&w=300&h=375",
        "title": "CANOPOLY",
        "about": "Fugiat velit commodo sint sit occaecat consequat magna nulla cupidatat incididunt magna. Veniam aliquip fugiat minim nulla incididunt aliqua dolore proident pariatur excepteur. Sunt pariatur enim minim magna. Do id dolor sint labore sint do eiusmod est et do ullamco aliquip.",
        "releaseDate": "January 6, 2014"
    }]