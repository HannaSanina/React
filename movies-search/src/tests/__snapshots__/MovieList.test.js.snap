// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieList/> should be rendered correctly 1`] = `
<div>
  <div
    className="resultsMovie"
  >
    <div
      className="\${styles.previewContainer} movieList"
    >
      <img
        alt="Movie"
        className="movie"
        src={undefined}
        title={undefined}
      />
      <div
        className="preview"
      >
        <span>
          CANOPOLY
        </span>
        <button>
          Order
        </button>
      </div>
    </div>
  </div>
</div>
`;
